File: Home/HalalEcodes/HalalEcodes.Data/Ecode.cs
using HalalEcodes.Data.Enums;
using HalalEcodes.Data.Models;

namespace HalalEcodes.Data
{
    public partial class Ecode : IEntity
    {
        public long Id { get; set; }
        public string Code { get; set; }
        public long CategoryId { get; set; }
        public string Ingredients { get; set; }
        public EcodeStatus Status { get; set; }
        public string StatusDesc { get; set; }
        public long EuApprouved { get; set; }
        public long UsApprouved { get; set; }
        public string MainIngredient { get; set; }
        public long ContainsAlcohol { get; set; }
        public long IsToxic { get; set; }

        public Category Category { get; set; }
    }
}

File: Home/HalalEcodes/HalalEcodes/ClientApp/app/models/ecode.ts
export class Ecode {
    id: number;
    code: string;
    category: string;
    ingredients: string;
    status: EcodeStatus;
    statusName: string;
    statusDesc: string;
    euApprouved: boolean;
    usApprouved: boolean;
    mainIngredient: string;
    containsAlcohol: boolean;
    isToxic: boolean;
}

export enum EcodeStatus {
    Halal = 0,
    Haram = 1,
    Mashbouh = 2,
    Unknown = 3
}